libs/template/docs/experimental/rx-for-directive.md

```diff
@@ -0,0 +1,211 @@
+
+This example showcases the `select` view-context function used for deeply nested lists.
+
+```html
+<ul>
+ <li *rxFor="let hero of heroes$; trackBy: trackItem; let select = select;">
+   <div>
+     <strong>{{ hero.name }}</strong></br>
+     Defeated enemies:
+   </div>
+    <span *rxFor="let enemy of select(['defeatedEnemies']); trackBy: trackEnemy;">
+      {{ enemy.name }}
+    </span>
+ </li>
+</ul>
+```
+
+### Input properties
+
+- trackBy: `(index: number, item: T) => any`
+- trackBy: `keyof T`
+- strategy: `string`
+- strategy: `Observable<string>`
+- parent: `boolean`;
+- renderCallback: `Subject<T[]>`
+
+### Using the context variables
+
+```html
+<ul>
+  <li
+    *rxFor="
+      let item of observableItems$;
+      let count = count;
+      let index = index;
+      let first = first;
+      let last = last;
+      let even = even;
+      let odd = odd;
+      trackBy: trackItem;
+    "
+  >
+    <div>{{ count }}</div>
+    <div>{{ index }}</div>
+    <div>{{ item }}</div>
+    <div>{{ first }}</div>
+    <div>{{ last }}</div>
+    <div>{{ even }}</div>
+    <div>{{ odd }}</div>
+  </li>
+</ul>
+```
+
+### Projected Views (`parent`)
+
+Imagine the following situation:
+
+```ts
+@Component({
+  selector: 'app-list-component',
+  template: ` <ng-content select="app-list-item"></ng-content> `,
+})
+export class AppListComponent {
+  @ContentChildren(AppListItemComponent)
+  appListItems: QueryList<AppListItemComponent>;
+}
+```
+
+`AppListComponent` has a `contentOutlet` where it expects `AppListItemComponents` to be inserted into. In this case `AppListComponent`s state is dependent on its `ContentChildren`.
+This situation leads to the problem that `AppListComponent` needs to get informed about updates of its child views. This is a known issue which has never been solved for `ngFor` (or other structural directives) especially in combination with `CD OnPush` see here: (https://github.com/angular/angular/pull/35428) `RxFor` solves this issue for you by providing a simple input parameter `parent: boolean`. If set to `true`, `*rxFor` will automatically detect every other `Component` where its `EmbeddedView`s were inserted into. Those components will get change detected as well in order to force update their state accordingly.
+
+The usage of `AppListComponent` looks like this:
+
+```html
+<app-list-component>
+  <app-list-item
+    *rxFor="
+      let item of observableItems$;
+      parent: true;
+    "
+  >
+    <div>{{ item }}</div>
+  </app-list-item>
+</app-list-component>
+```
+
+### `NgZone` patch
+
+By default `*rxFor` will create it's `EmbeddedViews` outside of `NgZone` which drastically speeds up the performance. There are scenarios where you want to opt-in to `NgZone` though. If views are created out of `NgZone`, all `EventListeners` attached to them run out `NgZone` as well.
```

Comment:
this aint true anymore, it will patch it into zone because everyone would get completely mad :D